src/main/java/net/consensys/orion/http/handler/privacy/PrivacyGroupBaseHandler.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package net.consensys.orion.http.handler.privacy;
+
+import static net.consensys.orion.http.server.HttpContentType.CBOR;
+
+import net.consensys.cava.crypto.sodium.Box;
+import net.consensys.orion.exception.OrionErrorCode;
+import net.consensys.orion.exception.OrionException;
+import net.consensys.orion.network.ConcurrentNetworkNodes;
+import net.consensys.orion.utils.Serializer;
+
+import java.io.Serializable;
+import java.net.URL;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClient;
+import io.vertx.ext.web.RoutingContext;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+abstract class PrivacyGroupBaseHandler {
+
+  private final Logger log = LogManager.getLogger();
+  private final ConcurrentNetworkNodes networkNodes;
+  private final HttpClient httpClient;
+
+  PrivacyGroupBaseHandler(ConcurrentNetworkNodes networkNodes, HttpClient httpClient) {
```

Comment:
was package private functions intended?

Reply:
As far as I can see all the handlers relating to privacy groups will be in this package, so package-private works here.